src/EventStore.Core/TransactionLog/Chunks/TFChunkWriter.cs

```diff
@@ -43,35 +45,45 @@ public class TFChunkWriter : ITransactionFileWriter {
 		public bool Write(ILogRecord record, out long newPos) {
 			var result = _currentChunk.TryAppend(record);
 			if (result.Success)
-				_writerCheckpoint.Write(result.NewPosition + _currentChunk.ChunkHeader.ChunkStartPosition);
+				_nextRecordPosition = result.NewPosition + _currentChunk.ChunkHeader.ChunkStartPosition;
 			else
 				CompleteChunk(); // complete updates checkpoint internally
-			newPos = _writerCheckpoint.ReadNonFlushed();
+			newPos = _nextRecordPosition;
 			return result.Success;
 		}
 
-		public void CompleteChunk() {
+		public void Commit() {
+			_writerCheckpoint.Write(_nextRecordPosition);
+		}
+
+		public void CompleteChunk(bool commitAndFlush = false) {
```

Comment:
hesitant about the default parameter value here and in CompleteReplicatedRawChunk, i think better to be explicit?

Reply:
fixed, it's no longer necessary now with Begin-Commit, we can commit inside CompleteChunk() if there is no ongoing transaction